modes.c

```diff
@@ -0,0 +1,63 @@
+#include <math.h>
+
+#include "pico/stdlib.h"
+#include "pico/cyw43_arch.h"
+#include "hardware/pwm.h"
+#include "modes.h"
+
+static const double pi = 3.14159;
```

Comment:
please, just use M_PI from <math.h>, it is more precise, and it is a good practice to use standard C libraries